pkg/kubelet/rkt/rkt.go

```diff
@@ -554,13 +554,15 @@ func setApp(imgManifest *appcschema.ImageManifest, c *api.Container, opts *kubec
 
 	// If 'User' or 'Group' are still empty at this point,
 	// then apply the root UID and GID.
-	// TODO(yifan): Instead of using root GID, we should use
-	// the GID which the user is in.
+	// TODO(yifan): If only the GID is empty, rkt should be able to determine the GID according
+	// to the /etc/passwd.
+	// See https://github.com/appc/spec/issues/623.
 	if app.User == "" {
 		app.User = "0"
+		app.Group = "0"
 	}
 	if app.Group == "" {
-		app.Group = "0"
+		return fmt.Errorf("cannot determine the GID of the app %q", imgManifest.Name)
```

Comment:
@tmrts the split between in-the-container users and on-the-host users means you won't get a sensical result at best, and in reality the user here will be `root` because the kubelet runs as root.